src/silo/storage/column_group.cpp

```diff
@@ -190,23 +194,39 @@ ColumnPartitionGroup ColumnPartitionGroup::getSubgroup(
    }
 
    for (const auto& item : fields) {
-      if (item.type == silo::config::ColumnType::INDEXED_STRING) {
-         result.indexed_string_columns.insert({item.name, indexed_string_columns.at(item.name)});
-      } else if (item.type == silo::config::ColumnType::STRING) {
-         result.string_columns.insert({item.name, string_columns.at(item.name)});
-      } else if (item.type == silo::config::ColumnType::INDEXED_PANGOLINEAGE) {
-         result.pango_lineage_columns.insert({item.name, pango_lineage_columns.at(item.name)});
-      } else if (item.type == silo::config::ColumnType::DATE) {
-         result.date_columns.insert({item.name, date_columns.at(item.name)});
-      } else if (item.type == silo::config::ColumnType::INT) {
-         result.int_columns.insert({item.name, int_columns.at(item.name)});
-      } else if (item.type == silo::config::ColumnType::FLOAT) {
-         result.float_columns.insert({item.name, float_columns.at(item.name)});
-      } else if (item.type == silo::config::ColumnType::NUC_INSERTION) {
-         result.nuc_insertion_columns.insert({item.name, nuc_insertion_columns.at(item.name)});
-      } else if (item.type == silo::config::ColumnType::AA_INSERTION) {
-         result.aa_insertion_columns.insert({item.name, aa_insertion_columns.at(item.name)});
-      }
+      ([&]() {
+         switch (item.type) {
+            case ColumnType::INDEXED_STRING:
+               result.indexed_string_columns.insert(
+                  {item.name, indexed_string_columns.at(item.name)}
+               );
+               return;
+            case ColumnType::STRING:
+               result.string_columns.insert({item.name, string_columns.at(item.name)});
+               return;
+            case ColumnType::INDEXED_PANGOLINEAGE:
+               result.pango_lineage_columns.insert({item.name, pango_lineage_columns.at(item.name)}
+               );
+               return;
+            case ColumnType::DATE:
+               result.date_columns.insert({item.name, date_columns.at(item.name)});
+               return;
+            case ColumnType::INT:
+               result.int_columns.insert({item.name, int_columns.at(item.name)});
+               return;
+            case ColumnType::FLOAT:
+               result.float_columns.insert({item.name, float_columns.at(item.name)});
+               return;
+            case ColumnType::NUC_INSERTION:
+               result.nuc_insertion_columns.insert({item.name, nuc_insertion_columns.at(item.name)}
+               );
+               return;
+            case ColumnType::AA_INSERTION:
+               result.aa_insertion_columns.insert({item.name, aa_insertion_columns.at(item.name)});
+               return;
+         }
+         abort();
+      })();
```

Comment:
Why was this lambda necessary?

Reply:
(For those reading later) In addition to the comments further above: lambda immediately called is a way (quite commonly used both in C++ and other languages) to enable use of `return` with a specified return boundary.